web/src/main/webapp/dataViews/tableView.ts

```diff
@@ -1109,6 +1130,36 @@ class PCASchemaReceiver extends OnCompleteRenderer<TableSummary> {
     }
 }
 
+/**
+ * Receives the result of a PCA computation and plots the singular values
+ */
+class SpectrumReceiver extends OnCompleteRenderer<EigenVal> {
```

Comment:
Maybe you can move this class to the SpectrumView class?
This file is large enough.